### ISSUE TITLE
Aria-Expanded is Missing from Accordion

### ISSUE THREAD

**Issue by frastlin** [IBM/carbon-components-react#1650]
## Detailed description

Hello, I am a screen reader user and was looking at the [code examples](http://react.carbondesignsystem.com/?selectedKind=Accordion&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) and the accordions do not properly handle the aria-expanded attribute. Also, it's [good practice to have accordions be button elements](https://inclusive-components.design/collapsible-sections/). Currently they are delineated as tabs which is really confusing because tabs and accordions are different. Tabs show information below the tab list and can be navigated and selected using the arrow keys. Accordions use the expanded and collapsed attribute and the content is inserted below the expanded element.
For a very good example of accordions, see [this example from AccDC](http://whatsock.com/tsg/Coding%20Arena/ARIA%20and%20Non-ARIA%20Accordions/Accordion%20(Internal%20Content)/demo.htm)

## Steps to reproduce the issue

1. Using [NVDA](https://www.nvaccess.org/download/) on Chrome Version 71.0.3578.80 (Official Build) (64-bit) visit the [accordion example](http://react.carbondesignsystem.com/?selectedKind=Accordion&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel)
2. hit enter on the first: "Expand/Collapse Section 1 title"

## What Happened

nothing. NVDA says nothing when enter is pressed on the top collapsible component.

## What Should Happen

NVDA should say "Expanded" and the next press of enter NVDA should announce "collapsed" (Although for some reason enter doesn't seem to collapse the component, it's probably because it's not a button and is not getting the key press).

**Comment by emyarod** [comment]
hi @frastlin thanks for reporting this issue! I've pushed a quick fix to add `aria-expanded` to the accordion [here](https://github.com/IBM/carbon-components-react/pull/1657), but I am still in the process of setting up NVDA on my machine. Can you confirm whether or not the component is improved at this [staging link](https://deploy-preview-1657--carbon-components-react.netlify.com/)? 

You also brought up a good point about our usage of the `tab` role right now, and while the pull request I referenced doesn't address it yet, it is also something I would like to explore further. But first I want to make sure that the component is detectable by screen readers after this change

**Comment by frastlin** [comment]
No, the [example](https://deploy-preview-1657--carbon-components-react.netlify.com/?selectedKind=Accordion&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel) does not work as expected.
The existing element looks like:

    <button type="button" class="bx--accordion__heading" role="tab"><svg class="bx--accordion__arrow" fill-rule="evenodd" height="12" role="img" viewBox="0 0 7 12" width="7" aria-label="Expand/Collapse" alt="Expand/Collapse"><title>Expand/Collapse</title><path d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"></path></svg><div class="bx--accordion__title">Section 1 title</div></button>

And should look like:

    <button type="button" class="bx--accordion__heading" aria-expanded="false"><svg class="bx--accordion__arrow" fill-rule="evenodd" height="12" role="img" viewBox="0 0 7 12" width="7" alt="Expand/Collapse"><title>Expand/Collapse</title><path d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"></path></svg><div class="bx--accordion__title">Section 1 title</div></button>

Note the removal of the role="tab" (It is not proper aria to make that button into a tab when a button is what it should be), and the addition of the aria-expanded="false" element. In the react code, the onClick function should change a Boolean state that says "expanded" and that should be in the aria-expanded.

    Aria-expanded={this.state.expanded}

Also, I don't think the aria-labeled in the svg does anything.

**Comment by frastlin** [comment]
Question:
Is the:

    <svg class="bx--accordion__arrow" fill-rule="evenodd" height="12" role="img" viewBox="0 0 7 12" width="7" aria-label="Expand/Collapse" alt="Expand/Collapse">

an arrow saying if the element can be expanded or collapsed? If so, it should be:

    <svg class="bx--accordion__arrow" fill-rule="evenodd" height="12" viewBox="0 0 7 12" width="7" aria-hidden="true">

From [this guide on tabs](https://inclusive-components.design/collapsible-sections/)

> Since state is communicated via the aria-expanded attribute non-visually, we don't need this graphic to be screen reader accessible or interactive.

Also, Accordion.js does not need:

    role="tablist"
    aria-multiselectable="true"

We don't need the tablist because this is an accordion, not a tab. We don't need the second one because we can't actually "select" the accordion element, instead we "expand" the element which is all performed with the aria-expanded attribute.

**Comment by emyarod** [comment]
hi @frastlin thank you for the feedback! It seems like some of my changes are in line with what you are suggesting, although I incorrectly placed them on the element that wraps the `<button>` instead of the `<button>` itself. The changes we can make to the markup in this repository are dictated by what we do in the [vanilla component repository](https://github.com/IBM/carbon-components). Is this issue also present in our vanilla accordion implementation? If so then I can make the necessary fixes to both repositories

**Comment by emyarod** [comment]
hi @frastlin thank you for the feedback! It seems like some of my changes were in line with what you are suggesting, although I incorrectly placed them on the element that wraps the <button> instead of the <button> itself. I've now made that correction, so would you be able to take another look at the [staging link](https://deploy-preview-1657--carbon-components-react.netlify.com/)?

Regarding your suggestions: I can definitely add the attributes you suggested and remove the linked attributes [from `Accordion.js`](https://github.com/IBM/carbon-components-react/blob/master/src/components/Accordion/Accordion.js#L13-L14) and the `role` attribute from the `<AccordionItem>` SVG icons. But can I get some more clarification on removing our existing `aria-label` and `alt` aria-attributes from the `<AccordionItem>` SVG icons?

**Comment by frastlin** [comment]
No, the [Vanilla component](http://www.carbondesignsystem.com/components/accordion/code) looks perfect.
What is the point of the svg? Why does it exist? If the answer is to tell the user that this is an expandable component and tell the state of the expanded component, then the aria-expanded attribute already does this.
Here is what NVDA says on aria-expanded="false" in the vanilla example:
"Button collapsed Section 1 title"
With aria-expanded="true" NVDA says:
"Button expanded Section 1 title"
Screen reader users know that button collapsed means the element is something they can hit enter or space on (enter and space both trigger the onClick event), and because the button with aria-expanded is such a common feature, hearing "expanded" or "collapsed" lets us know the state and that we can change the state. Also, the aria-expanded attribute has NVDA announce "Expanded" or "collapsed" when the state changes.
Adding the svg just confuses what is going on. Here is what NVDA would say if that svg had the alt description, img role and was not hidden:
"Button collapsed graphic Expand/Collapse Section 1 title"
"Button expanded graphic Expand/Collapse Section 1 title"
In contrast, hiding the svg would be:
"Button collapsed Section 1 title"
The first option has lots of extra information and the second one has just the information we need. In UX, this second option follows two [heuristics](https://www.nngroup.com/articles/ten-usability-heuristics/):

1. Consistency and standards: button combined with aria-expanded is an accordion to screen reader users.
2. Aesthetic and minimalist design: The "graphic Expand/Collapse" distracts from the important information of element, state, title.

(Also if aria-hidden="true" is on any element, it is hidden from the screen reader, so those extra accessibility elements are just taking up space)
The [latest fix](https://deploy-preview-1657--carbon-components-react.netlify.com/?selectedKind=Accordion&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel) looks great thank you!

**Comment by emyarod** [comment]
thank you for the clarification @frastlin ! yes I deleted and reposted my response once I realized what the issue was. glad to hear that the latest fix has improved things!
